src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -162,13 +172,14 @@ public void removeCommission(Customer customer, Commission key) {
         requireAllNonNull(customer, key);
         assert hasCommission(customer, key);
         // make a copy of customer's commissions with key removed.
-        Set<Commission> filteredCommissions = getCustomerCommissions(customer).stream()
-                .filter(commission -> !commission.equals(key)).collect(Collectors.toSet());
+        //Set<Commission> filteredCommissions = getCustomerCommissions(customer).stream()
+        //        .filter(commission -> !commission.equals(key)).collect(Collectors.toSet());
         commissions.remove(key);
         // make a copy of customer with the updated commissions.
-        Customer newCustomer = customer.copyWithCommissions(filteredCommissions);
+        // Customer newCustomer = customer.copyWithCommissions(filteredCommissions);
         // replace original customer with updated customer to trigger customer list UI updates
-        setCustomer(customer, newCustomer);
+        // setCustomer(customer, newCustomer);
+        customer.getCommissionsMutable().remove(key);
```

Comment:
Hmm, I modified this because originally it will cause a bug which after deleting commission A, then subsequently trying to add commission A a duplicate commission error is shown. I think it is because selectedCustomer still references the old customer even though a new customer replaced the instance. I don't think it is necessary to replace a customer with the deletion command since the observable list can detect an object being removed. More efficient also? 

P.S. I added another getter in customer class to enable mutability, don't know if that's the best.

Reply:
There might be issues with exposing the commission list of the customer like this in the long run since this would allow anyone to edit it freely, but it can be fixed in #82 when `removeCommission` is moved into `Customer`